coordinator/src/main/scala/filodb.coordinator/queryplanner/DefaultPlanner.scala

```diff
@@ -408,46 +408,67 @@ trait  DefaultPlanner {
     PlanResult(Seq(scalarBinaryExec))
   }
 
+  /**
+   * Function to optimize and convert `foo or vector(0)` like queries which would normally
+   * result in an expensive SetOperatorExec plan into a simple InstantFunctionMapper
+   * which is far more efficient.
+   */
+  def optimizeOrVectorZero(qContext: QueryContext, logicalPlan: BinaryJoin): Option[PlanResult] = {
+    if (logicalPlan.operator == BinaryOperator.LOR) {
+      logicalPlan.rhs match {
```

Comment:
This optimization would work only when vector(0) if RHS like in ``foo or vector(0)``? If someone does ``vector(0) or foo`` that still would be a ``SetOperatorExec``?  Something like [this](http://demo.robustperception.io:9090/graph?g0.expr=vector(0)%20or%20up&g0.tab=1&g0.stacked=0&g0.range_input=1h&g1.expr=up%20or%20vector(0)&g1.tab=1&g1.stacked=0&g1.range_input=1h) for example

Reply:
I left the complexity (and test case) out because it is an uncommon query. Let us cross the bridge when really needed. Default impl should suffice for that case.

Reply:
Can we call that out in comments in that case? Though uncommon, It will generate a different plan and someone working  on it in future will remember this caveat and why we deliberately left it out.